crates/sui-source-validation-service/src/lib.rs

```diff
@@ -361,17 +377,25 @@ pub async fn verify_packages(config: &Config, dir: &Path) -> anyhow::Result<Netw
             PackageSource::Repository(r) => {
                 let repo_name = repo_name_from_url(&r.repository)?;
                 let network_name = r.network.clone().unwrap_or_default().to_string();
-                let packages_dir = dir.join(network_name).join(repo_name);
-                for p in &r.packages {
-                    let package_path = packages_dir.join(p.path.clone()).clone();
-                    let network = r.network.clone().unwrap_or_default();
-                    let t =
-                        tokio::spawn(async move { verify_package(&network, package_path).await });
-                    tasks.push(t)
+                for b in &r.branches {
+                    for p in &b.paths {
+                        let sanitized_branch = b.branch.replace('/', "__");
+                        let package_path = dir
+                            .join(network_name.clone())
+                            .join(format!("{repo_name}__{sanitized_branch}"))
+                            .join(p.path.clone())
+                            .clone();
+                        let network = r.network.clone().unwrap_or_default();
+                        let t =
+                            tokio::spawn(
+                                async move { verify_package(&network, package_path).await },
+                            );
```

Comment:
Out of interest, is there a difference between what is written here and the following:

```suggestion
                            tokio::spawn(verify_package(&network, package_path));
```

I am not familiar enough with the intricacies of Rust's `async`/`await` to know myself, but I know that @stefan-mysten and I spotted something strange in how the CLI handles RPC calls (some strange race condition) and in that   case, the code doesn't wrap the async function call in an `async move` closure, and now I'm wondering whether that does something important. (@stefan-mysten , maybe it's worth checking whether this pattern would fix our issue with `rpc.discover` tripping up future RPC calls in the CLI).

Reply:
@amnn thanks for the hint, I'll try it asap.